chunk_utils.py

```
@@ -0,0 +1,209 @@
+from datetime import datetime
+from pathlib import Path
+from typing import List, Dict, Any, Optional, Union, Tuple
+from urllib.parse import urlparse
+
+
+@dataclass
+class Chunk:
+    """Represents a chunk of content for RAG processing."""
+    id: str
+    content: str
+    metadata: Dict[str, Any]
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)